frontend/model/contracts/group.js

```diff
@@ -727,6 +734,20 @@ DefineContract({
         }
       }
     },
+    'gi.contracts/group/updateVotingRules': {
+      validate: objectOf({
+        ruleName: x => [RULE_PERCENTAGE, RULE_DISAGREEMENT].includes(x),
+        ruleThreshold: number
+      }),
+      process ({ data, meta }, { state }) {
+        // Update all types of proposal settings for simplicity
+        for (const ruleName in state.settings.proposals) {
+          Vue.set(state.settings.proposals[ruleName], 'rule', data.ruleName)
+          Vue.set(state.settings.proposals[ruleName].ruleSettings[data.ruleName], 'threshold', data.ruleThreshold)
```

Comment:
Things should be named after what they do. If all this does is update the threshold for a rule, then the name should reflect that.

`'gi.contracts/group/updateVotingRules'` => `'gi.contracts/group/updateVotingRuleThreshold'`

Reply:
I only update the threshold because it's the only thing we need ATM. But it can be modified to support other things too. (e.g. `expires_ms`). I just didn't do it now because it's not needed.

Reply:
Remember, nothing here “can be modified”. These functions are set in stone after the prototype is released, so that they generate the same exact state forever.

Reply:
I understand that. Before the prototype is released, we will still touch this again to do #620, so I didn't think it was a big deal for now. But I'll fix it :)